src/ArcGISRuntime.Samples.Shared/Models/SampleModel.cs

```diff
@@ -204,9 +222,15 @@ internal static SampleModel Create(string metadataFilePath)
                 sampleModel.SampleFolder = metadataFile.Directory;
             }
 
+            // Stop if the sample doesn't have a VB implementation (VB sample viewer only)
+            if (ApplicationManager.Current.SelectedLanguage == Language.VBNet && System.Type.GetType(sampleModel.SampleVbTypeName, false) == null)
```

Comment:
You have the sample's folder available, could you just check the \<samplename\>VB.xaml exists? if not return